message/FieldMessageSerializer.h

```diff
@@ -0,0 +1,22 @@
+#ifndef BATTLECITY_FIELDMESSAGESERIALIZER_H
+#define BATTLECITY_FIELDMESSAGESERIALIZER_H
+
+#include "FieldMessage.h"
+
+class FieldMessageSerializer {
+
+public:
+    static constexpr int8_t INT32_TYPE_CODE = 0;
+    static constexpr int8_t STRING_TYPE_CODE = 127;
+
+public:
+    FieldMessageSerializer() = delete;
+
+public:
+    static std::string Serialize(const FieldMessage &message);
+
+    static FieldMessage Deserialize(std::string &&serialized);
```

Comment:
Taking rvalue values is an exceptional case. (`std::string&&` will confuse people). Remember the simple rules:
- you want to only read data:
```cxx
auto first(std::string_view str) -> char  {
    return str[0];
}
```
**do not use constant references to owned structures in this case**
~~const std::string&~~ should work with allocated on heap strings, string literals, etc. 

- you want to delegate value into something (insert to map, ...):
```cxx
void foo(std::string str){
    buf.insert(std::move(str));
}
```
take `std::string` by value and move into something
in this case if the string is already allocated, you won't need to copy it. 
In addition, you determines whether to move the string 
```cxx
foo("lol"); // create new `std::string` and immediately  move into foo

std::string x = "kek";
foo(x); // copy `x` into foo
foo(std::move(x)); // move `x` into foo
```

- you want to delegate **any** template type:
```cxx
auto invoke(auto&& foo, auto&&... args) -> decltype(auto) {
    return foo(std::forward<decltype(args)>(args)...);
}
```
use `std::forward` to **forward** kind of value next

Reply:
`FieldMessageSerializer::Serialize()` returns `std::string` in this format:
`9\000\000\000\000\000\000\000?\000\000\000\000\000\000\000\177\b\000Vladimir\177\005\000\062\062;58\177\004\000DOWN\000\377\000\000\000\000�\003\000\000\000\003\000\000`.
Constructor of `std::ostringstream` class can't receive `std::string_view` object as parameter, and `std::string_view::data()` (`std::string::data()` too) in this example returns `9` here because after the first character goes `'\0'`. So that I used `std::string` exactly, and just don't know how to optimize this method using `std::string_view`, so copying by value or reference in my opinion is the only way.

Reply:
Ok. Misstakes in `std`. You can manually create `std::string` from pointer to `.data()` and size
Or recieve parameter as `std::string`

Reply:
Or don't use `std::istringstream`. I serialize from pure `std::string_view`.